custom_components/wyzeapi/sensor.py

```diff
@@ -215,3 +244,219 @@ def extra_state_attributes(self):
     @property
     def native_value(self):
         return self._camera.device_params.get("electricity")
+
+
+class WyzePlugEnergySensor(RestoreSensor):
+    """Respresents an Outdoor Plug Total Energy Sensor."""
+
+    _attr_device_class = SensorDeviceClass.ENERGY
+    _attr_native_unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
+    _attr_state_class = SensorStateClass.TOTAL_INCREASING
+    _attr_suggested_display_precision = 3
+    _previous_hour = None
+    _previous_value = None
+    _past_hours_previous_value = None
+    _current_value = 0
+    _past_hours_value = 0
+    _hourly_energy_usage_added = 0
+
+    def __init__(
+        self, switch: Switch, switch_usage_service: SwitchUsageService
+    ) -> None:
+        """Initialize an energy sensor."""
+        self._switch = switch
+        self._switch_usage_service = switch_usage_service
+        self._switch.usage_history = None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the sensor."""
+        return "Total Energy Usage"
+
+    @property
+    def unique_id(self):
+        """Get the unique ID of the sensor."""
+        return f"{self._switch.nickname}.energy-{self._switch.mac}"
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._switch.mac)},
+            "name": self._switch.nickname,
+        }
+
+    def update_energy(self):
+        """Update the energy sensor."""
+        _now = int(datetime.utcnow().hour)
+        self._hourly_energy_usage_added = 0
+
+        if self._switch.usage_history:
+            _raw_data = self._switch.usage_history
+            _LOGGER.debug(_raw_data)
+            _current_day_list = json.loads(_raw_data[0]["data"])
```

Comment:
For some clarity, this is what's returned here: 

`{'date_ts': 1670457600000, 'data': '[83,83,76,82,3,0,0,0,0,0,0,148,163,49,0,0,39,0,0,0,0,11,166,153]'}`

The data block is 24 hours of usage data, and it's length is always 24 slices, regardless of the time of day. So as long as we have data, (which is checked in line 299 and again in 304, the other indices should always have data, either actual usage or a 0.